### PR TITLE
Add Storage.Close method

### PR DESCRIPTION
**Pull request by rojer9-fb**
To ensure all pending events are flushed when server shuts down.

Signed-off-by: Deomid "rojer" Ryabkov <rojer9@fb.com>

### CODE REVIEW

File: cmds/contest/server/server.go [hunk 1]
```diff
@@ -221,6 +231,14 @@ func ServerMain(cmd string, args []string, sigs <-chan os.Signal) error {
 
 	err = jm.Run(ctx, *flagResumeJobs)
 
+	target.SetLocker(nil)
+	if err := storage.SetStorage(nil); err != nil {
+		log.Errorf("Failed to flush storage")
+	}
+	if err := storage.SetAsyncStorage(nil); err != nil {
+		log.Errorf("Failed to flush async storage")
+	}
```
Comment (tfg13) [inline cmds/contest/server/server.go:240]: From how I read this, this will never error with `nil` as a storage engine

Comment (rojer9-fb) [reply cmds/contest/server/server.go:240]: yes, and this is fine: if user want to disable storage, they should be allowed to. we do it when tearing down tests.

Comment (tfg13) [reply cmds/contest/server/server.go:240]: Yeah I get the disable usecase, but not sure what we are printing here, as the log lines can never be reached?

**Review by tfg13 — APPROVED** [review]

File: cmds/contest/server/server.go (hunk 1 above)
Comment (rojer9-fb) [reply cmds/contest/server/server.go:240]: oh yeah, but lint will complain if i don't check the return value :)
but i guess i can simplify it by using a sentinel

**Pull request merged**